datasource/src/main/scala/quasar/blobstore/azure/AzureConfig.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.blobstore.azure
+
+import slamdata.Predef._
+
+final case class ContainerName(value: String)
+final case class StorageUrl(value: String)
+
+final case class AccountName(value: String)
+final case class AccountKey(value: String)
+
+final case class AzureCredentials(accountName: AccountName, accountKey: AccountKey)
+
+final case class AzureConfig(containerName: ContainerName, credentials: Option[AzureCredentials], storageUrl: StorageUrl)
```

Comment:
Minor point… I prefer to give files lowercase and more general names when containing multiple top-level compilation units (like this one). So rather than naming it `AzureConfig.scala`, which kind of implies that it *only* contains a top-level `AzureConfig` class and/or object, instead call it something like `config.scala`. This is a pretty widespread convention and makes the codebase much easier to browse for people who aren't using ctags, an IDE, or sublime-style symbol indexing.